packages/datadog-plugin-aws-sdk/src/services/sqs.js

```diff
@@ -78,24 +78,39 @@ class Sqs extends BaseAwsSdkPlugin {
   }
 
   responseExtract (params, operation, response) {
-    if (operation === 'receiveMessage') {
-      if (
-        (!params.MaxNumberOfMessages || params.MaxNumberOfMessages === 1) &&
-        response &&
-        response.Messages &&
-        response.Messages[0] &&
-        response.Messages[0].MessageAttributes &&
-        response.Messages[0].MessageAttributes._datadog &&
-        response.Messages[0].MessageAttributes._datadog.StringValue
-      ) {
-        const textMap = response.Messages[0].MessageAttributes._datadog.StringValue
-        try {
-          return this.tracer.extract('text_map', JSON.parse(textMap))
-        } catch (err) {
-          log.error(err)
-          return undefined
+    if (operation !== 'receiveMessage') return
+    if (params.MaxNumberOfMessages && params.MaxNumberOfMessages !== 1) return
+    if (!response || !response.Messages || !response.Messages[0]) return
+
+    let message = response.Messages[0]
+
+    if (message.Body) {
+      try {
+        const body = JSON.parse(message.Body)
+
+        // SNS to SQS
+        if (body.Type === 'Notification') {
+          message = body
         }
+      } catch (e) {
+        // SQS to SQS
```

Comment:
Do we want a debug log here?

Reply:
At what log level? For errors, log level should generally be error since it's the only one that support actual error objects, but in this case it's an expected error in the sense that it might just mean it's not SNS to SQS. I thought of checking for the presence of `"Type": "Notification"` in the string instead of just parsing right away to avoid this, but ultimately decided to leave it to the try/catch. Not sure what approach would be best here as I'm not familiar with all the possible ways this response can be structured.

Reply:
Would say debug, but actually this is probably not important.

Type: Notification should work, but in THEORY some user could be manually adding that to their own SQS payload for some reason.

perhaps `TopicArn` being present, plus `body.Type === 'Notification'` would be better?

I'm also happy to leave this as-is.

Reply:
Let's leave it for now, at least we know it's guaranteed to work.